src/v/transform/logging/rpc_client.cc

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "rpc_client.h"
+
+#include "cluster/topic_table.h"
+#include "model/namespace.h"
+#include "transform/logging/logger.h"
+#include "transform/logging/record_batcher.h"
+
+#include <seastar/core/future.hh>
+
+namespace transform::logging {
+
+rpc_client::rpc_client(
+  rpc::client* rpc_client, cluster::topic_table* topic_table)
+  : _rpc_client(rpc_client)
+  , _topic_table(topic_table) {}
+
+ss::future<errc>
+rpc_client::write(model::partition_id pid, io::json_batches batches) {
+    model::topic_namespace_view ns_tp{model::transform_log_internal_nt};
+    // TODO(oren): is topic metadata cache OR topic table more preferable here?
```

Comment:
Probably the topic metadata cache is fine here. Can we just pass in the metadata cache here and not expose this in the rpc client? Feels like a violation of responsibility a bit.

Reply:
done